libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -677,12 +721,13 @@ void AP_GPS::update_instance(uint8_t instance)
         if (tnow - timing[instance].last_message_time_ms > GPS_TIMEOUT_MS) {
             memset((void *)&state[instance], 0, sizeof(state[instance]));
             state[instance].instance = instance;
+            state[instance].devid = _devid[instance];
             state[instance].hdop = GPS_UNKNOWN_DOP;
             state[instance].vdop = GPS_UNKNOWN_DOP;
             timing[instance].last_message_time_ms = tnow;
             timing[instance].delta_time_ms = GPS_TIMEOUT_MS;
-            // do not try to detect again if type is MAV
-            if (_type[instance] == GPS_TYPE_MAV) {
+            // do not try to detect again if type is MAV or UAVCAN
+            if (_type[instance] == GPS_TYPE_MAV || _type[instance] == GPS_TYPE_UAVCAN) {
```

Comment:
does this affect CAN failover? Say for example a GPS disappears off one CAN bus and appears on the other. I haveb't actually tested this, but given we store an AP_UAVCAN bus instance in the driver then I'm concerned this will break failover between CAN buses. Shouldn't be hard to test, just have GPS_TYPE=9, GPS_TYPE2=0, attach CAN GPS to both CAN1 and CAN2, then alternately disconnect one or the other. With this change I'm concerned we'd lose the GPS